pubsub/driver/driver.go

```diff
@@ -0,0 +1,46 @@
+package driver
+
+import "context"
+
+type AckID interface{}
+
+type Message struct {
+	// Body contains the content of the message.
+	Body []byte
+
+	// Attributes has key/value metadata for the message.
+	Attributes map[string]string
+
+	// AckID identifies the message on the server. It can be used to ack
+	// the message after it has been received.
+	AckID AckID
+
+	// ErrChan relays back an error or nil as the result of sending the
+	// batch that includes this message.
+	ErrChan chan error
+}
+
+// Topic publishes messages.
+type Topic interface {
+	// SendBatch publishes all the messages in ms.
```

Comment:
Does it also write to ErrChan for each of the messages?

Reply:
It doesn't. I'm pretty sure it was a mistake having the ErrChan field on driver.Message. Most likely the errChans can be fields of little structs like this that get sent to the message batcher:

```go
type msgErrChan struct {
    msg *Message
    errChan chan error
}
```
That should keep the Message structs cleaner.